src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react';
+import produce from 'immer';
+
+import VirtualMachineModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineModel';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  Button,
+  ButtonVariant,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  Modal,
+  ModalVariant,
+  NumberInput,
+  Title,
+  TitleSizes,
+} from '@patternfly/react-core';
+
+import useTemplateDefaultCpuMemory from './hooks/useTemplateDefaultCpuMemory';
+import { getCPUCount, getMemorySize, memorySizesTypes } from './utils/CpuMemoryUtils';
+
+import './cpu-memory-modal.scss';
+
+type CPUMemoryModalProps = {
+  vm: V1VirtualMachine;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+const CPUMemoryModal: React.FC<CPUMemoryModalProps> = ({ vm, isOpen, onClose }) => {
+  const { t } = useKubevirtTranslation();
+  const {
+    data: templateDefaultsData,
+    loaded: defaultsLoaded,
+    error: defaultLoadError,
+  } = useTemplateDefaultCpuMemory(vm);
+  const [updateInProcess, setUpdateInProcess] = React.useState<boolean>(false);
+  const [updateError, setUpdateError] = React.useState<string>();
+
+  const [memory, setMemory] = React.useState<number>();
+  const [cpuCores, setCpuCores] = React.useState<number>();
+  const [memoryUnit, setMemoryUnit] = React.useState<string>();
+  const [isDropdownOpen, setIsDropdownOpen] = React.useState<boolean>(false);
+
+  const resultVirtualMachine = React.useMemo(() => {
+    const updatedVM = produce<V1VirtualMachine>(vm, (vmDraft: V1VirtualMachine) => {
+      vmDraft.spec.template.spec.domain.resources.requests = {
+        ...vm?.spec?.template?.spec?.domain?.resources?.requests,
```

Comment:
why do u need this spread? `immer` takes care of that.
I would also use `ensurePath` to make sure the path is existing.

```
const updatedVM = produce<V1VirtualMachine>(vm, (vmDraft: V1VirtualMachine) => {
    ensurePath(vmDraft, ['spec.template.spec.domain.resources.requests', 'spec.template.spec.domain.cpu'])

    vmDraft.spec.template.spec.domain.resources.requests.memory = `${memory}${memoryUnit}`;
    vmDraft.spec.template.spec.domain.cpu.cores = cpuCores;
});

Reply:
`resources` is of type object or to be exact: `{}` like `any` so `requests.memory` is not recognized as a type,
had to do it to keep previous values if there are any